scripts/validate-packages.py

```diff
@@ -77,14 +79,26 @@ def _validate_revision(given_package, revision, path):
 
     # validate resource.json
     resource_json_path = os.path.join(path, 'resource.json')
+    resource_json = None
     if os.path.isfile(resource_json_path):
         print("\t\tresource.json:", end='')
         if packaging_version == "2.0":
-            _validate_json(resource_json_path, V2_RESOURCE_JSON_SCHEMA)
+            resource_json = _validate_json(
+                resource_json_path,
+                V2_RESOURCE_JSON_SCHEMA)
         else:
-            _validate_json(resource_json_path, V3_RESOURCE_JSON_SCHEMA)
+            resource_json = _validate_json(
+                resource_json_path,
+                V3_RESOURCE_JSON_SCHEMA)
         print("\tOK")
 
+    # Validate that we don't drop information during the conversion
+    oldPackage = LooseVersion(
+        package_json.get('minDcosReleaseVersion', "1.0")) < LooseVersion("1.8")
+    if oldPackage and 'cli' in resource_json and command_json is None:
+        sys.exit('\tERROR\n\nWill drop CLI information if converted to '
+                 'version 2 package. Please set the minDcosReleaseVerions to '
+                 '1.8.')
```

Comment:
I'm not sure package maintainers will understand the `if converted to version 2 package` part of the message. Maybe something like "binary CLIs areonly available for DCOS version >= 1.8. To support CLIs in earlier DCOS versions please specify a pip installable CLI in command.json"